### DISCUSSION TITLE
file-specific config?

### DISCUSSION THREAD

**Discussion by Jasha10** [microsoft/pyright#2603] [Ideas]
Hi,

Is it possible, configure pyright with file-specific settings?
I'm aware that the `ignore` config option allows for checking of specific files to be disabled... But is it possible to e.g. ignore just one type of error for specific files?

**Comment by erictraut** [comment]
Yes, you can do this through the use of a `# pyright` comment within the file. For details refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/comments.md).

If you are running the command-line version of pyright, you also have the option of specifying different config files that run on different parts of your code base. You could then write a script that invokes pyright multiple times, each time specifying a different config file. Each config file can specify a different set of source files (through the `include` setting) and enable different sets of diagnostic rules.

**Reply by Jasha10** [reply]
Got it. Thanks very much!
